go/libraries/doltcore/diff/async_differ.go

```diff
@@ -91,13 +118,97 @@ func (ad *AsyncDiffer) GetDiffs(numDiffs int, timeout time.Duration) ([]*diff.Di
 	}
 }
 
-func (ad *AsyncDiffer) ReadAll() ([]*diff.Difference, error) {
-	diffs, hasMore, err := ad.GetDiffs(0, 5*time.Minute)
-	if err != nil {
-		return nil, err
+type keylessDiffer struct {
+	*AsyncDiffer
+
+	df         diff.Difference
+	copiesLeft uint64
+}
+
+var _ RowDiffer = &keylessDiffer{}
+
+func (kd *keylessDiffer) GetDiffs(numDiffs int, timeout time.Duration) (diffs []*diff.Difference, more bool, err error) {
+	timeoutChan := time.After(timeout)
+	diffs = make([]*diff.Difference, numDiffs)
+	idx := 0
+
+	for {
+		// first populate |diffs| with copies of |kd.df|
+		for (idx < numDiffs) && (kd.copiesLeft > 0) {
+			diffs[idx] = &kd.df
+
+			idx++
+			kd.copiesLeft--
+		}
+		if idx == numDiffs {
+			return diffs, true, nil
+		}
+
+		// then get another Difference
+		var d diff.Difference
+		select {
+		case <-timeoutChan:
+			return diffs, true, nil
+
+		case <-kd.egCtx.Done():
+			return nil, false, kd.eg.Wait()
+
+		case d, more = <-kd.diffChan:
+			if !more {
+				return diffs[:idx], more, nil
+			}
+
+			kd.df, kd.copiesLeft, err = convertDiff(d)
```

Comment:
Bet you're glad we talked you into using compatible noms storage now huh

Reply:
Coulda worked either way, consumers of `GetDiffs()` use `row.FromNoms` and then work with `row.Row`s